### PR TITLE
HPCC-26050 Logical File details page Superfiles tab

### PR DESCRIPTION
**Pull request by jeclrsg**
Signed-off-by: Jeremy Clements <jeremy.clements@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/routes.tsx [hunk 1]
```diff
@@ -72,6 +72,7 @@ const routes: Routes = [
         children: [
             { path: "", action: (context) => import("./components/Files").then(_ => <_.Files filter={parseSearch(context.search) as any} />) },
             { path: "/legacy", action: () => import("./layouts/DojoAdapter").then(_ => <_.DojoAdapter widgetClassID="DFUQueryWidget" />) },
+            { path: "/:Name", action: (ctx, params) => import("./components/FileDetails").then(_ => <_.FileDetails logicalFile={params.Name as string} />) },
```
Comment (GordonSmith) [inline esp/src/src-react/routes.tsx:75]: This won't work with the "tabs" paradigm as you end up with `/:Name/:Tab` which is the same as `/:NodeGroup/:Name`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: esp/src/src-react/components/FileDetails.tsx [hunk 2]
```diff
@@ -148,7 +149,8 @@ export const FileDetails: React.FunctionComponent<FileDetailsProps> = ({
             </PivotItem>
             <PivotItem headerText={nlsHPCC.XML} itemKey="XML" style={pivotItemStyle(size, 0)}>
             </PivotItem>
-            <PivotItem headerText={nlsHPCC.Superfiles} itemKey="Superfiles" style={pivotItemStyle(size, 0)}>
+            <PivotItem headerText={nlsHPCC.Superfiles} itemKey="superfiles" itemCount={file?.Superfiles?.DFULogicalFile.length || 0} style={pivotItemStyle(size, 0)}>
```
Comment (jeclrsg) [inline esp/src/src-react/components/FileDetails.tsx:152]: Noticed this when making changes from the previous review. The || 0 should be an ok default, if something is undefined?

Comment (GordonSmith) [reply esp/src/src-react/components/FileDetails.tsx:162]: Yes that is fine.

Comment (GordonSmith) [reply esp/src/src-react/components/FileDetails.tsx:162]: (As long as you want (0) to be displayed?

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**